synapseutils/walk.py

```diff
@@ -1,8 +1,24 @@
 import synapseclient
 from synapseclient.entity import is_container
 import os
+import json
 
-def walk(syn, synId):
+def _getChildren(syn, parentId, includeTypes=["folder","file","table","link","entityview","dockerrepo"], sortBy="NAME", sortDirection="ASC"):
+    entityChildrenRequest = {'parentId':parentId,
+                             'includeTypes':includeTypes,
+                             'sortBy':sortBy,
+                             'sortDirection':sortDirection,
+                             'nextPageToken':None}
+    allChildren = []
+    resultPage = syn.restPOST('/entity/children',body =json.dumps(entityChildrenRequest))
+    allChildren = resultPage['page']
+    while resultPage.get('nextPageToken') is not None:
+        entityChildrenRequest['nextPageToken'] = resultPage['nextPageToken']
+        resultPage = syn.restPOST('/entity/children',body =json.dumps(entityChildrenRequest))
+        allChildren.extend(resultPage['page'])
```

Comment:
Tom:  To make this an iterator (whcih I recomnend) you need to make a for loop within your while loop where you yield the individual results in your resultPage['page']